src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -0,0 +1,289 @@
+
+NamesAndTypesList StorageSystemDDLWorkerQueue::getNamesAndTypes()
+{
+    return {
+        {"entry", std::make_shared<DataTypeString>()},
+        {"host_name", std::make_shared<DataTypeString>()},
+        {"host_address", std::make_shared<DataTypeString>()},
+        {"port", std::make_shared<DataTypeUInt16>()},
+        {"status", std::make_shared<DataTypeEnum8>(getStatusEnumsAndValues())},
+        {"cluster", std::make_shared<DataTypeString>()},
+        {"value", std::make_shared<DataTypeString>()},
+        {"query_start_time", std::make_shared<DataTypeDateTime>()},
+        {"query_finish_time", std::make_shared<DataTypeDateTime>()},
+        {"query_duration_ms", std::make_shared<DataTypeUInt64>()},
+        {"exception_code", std::make_shared<DataTypeEnum8>(getZooKeeperErrorEnumsAndValues())},
+    };
+}
+
+
+static bool extractPathImpl(const IAST & elem, String & res, const Context & context)
+{
+    const auto * function = elem.as<ASTFunction>();
+    if (!function)
+        return false;
+
+    if (function->name == "and")
+    {
+        for (const auto & child : function->arguments->children)
+            if (extractPathImpl(*child, res, context))
+                return true;
+        return false;
+    }
+
+    if (function->name == "equals")
+    {
+        const auto & args = function->arguments->as<ASTExpressionList &>();
+        ASTPtr value;
+
+        if (args.children.size() != 2)
+            return false;
+
+        const ASTIdentifier * ident;
+        if ((ident = args.children.at(0)->as<ASTIdentifier>()))
+            value = args.children.at(1);
+        else if ((ident = args.children.at(1)->as<ASTIdentifier>()))
+            value = args.children.at(0);
+        else
+            return false;
+
+        if (ident->name() != "cluster")
+            return false;
+
+        auto evaluated = evaluateConstantExpressionAsLiteral(value, context);
+        const auto * literal = evaluated->as<ASTLiteral>();
+        if (!literal)
+            return false;
+
+        if (literal->value.getType() != Field::Types::String)
+            return false;
+
+        res = literal->value.safeGet<String>();
+        return true;
+    }
+    return false;
+}
+
+
+/** Retrieve from the query a condition of the form `path = 'path'`, from conjunctions in the WHERE clause.
+  */
+static String extractPath(const ASTPtr & query, const Context & context)
+{
+    const auto & select = query->as<ASTSelectQuery &>();
+    if (!select.where())
+        return "";
+    String res;
+    return extractPathImpl(*select.where(), res, context) ? res : "";
+}
+
+void StorageSystemDDLWorkerQueue::fillData(MutableColumns & res_columns, const Context & context, const SelectQueryInfo & query_info) const
+{
+    String cluster_name = extractPath(query_info.query, context);
+    if (cluster_name.empty())
+        throw Exception(
+            "SELECT from system.ddl_worker_queue table must contain condition like cluster = 'cluster' in WHERE clause.",
```

Comment:
Why is it necessary? Maybe show queries for all clusters? If query contains WHERE clause with condition on cluster name, all irrelevant rows will be filtered later in query pipeline.

Reply:
Sure, I can change.

Reply:
Showing all clusters and the necessary clusters can be filtered later in the pipeline as suggested.